gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -138,26 +138,23 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
   }
 
   /*
-    validates if lease can be acquired on the provided flowSpec,
-    else throw LeaseUnavailableException
+    enforces that a similar flow is not launching,
+    else throw TooSoonToRerunSameFlowException
    */
-  private void validateAdhocFlowLeasability(FlowSpec flowSpec) {
+  private void enforceSimilarAdhocFlowExistence(FlowSpec flowSpec) {
```

Comment:
this doesn't enforce the existence, but rather the **non-**existence.

also, "similar lease" generally sounds apt, if we consider "similar" to be **the same** FlowId, but different executionId.  hence the lease is similar, while "the flow" is actually... **the same**.

this line of reasoning leaves "similar flow" sounding imprecise at best and confusing at worst.  I regret suggesting it and apologize for that.  (it stands out more clearly when I'm solely reading vs. struggling to originate a name myself.)

really, we're talking about recent execution of the same FlowId (aka. flowGroup+flowName).  maybe `enforceNoRecentAdhocExecOfSameFlow`?

Reply:
sure, updated